WordPress/src/main/java/org/wordpress/android/ui/stats/ReferrerSpamHelper.java

```diff
@@ -0,0 +1,159 @@
+                return true;
+            }
+        });
+
+        popup.show();
+    }
+
+
+    private class ReferrerSpamRestListener implements RestRequest.Listener, RestRequest.ErrorListener {
+        private final WeakReference<Activity> mActivityRef;
+        private final ReferrerGroupModel mReferrerGroup;
+        private final boolean isMarkingAsSpamInProgress;
+
+        public ReferrerSpamRestListener(Activity activity, final ReferrerGroupModel referrerGroup, final boolean isMarkingAsSpamInProgress) {
+            this.mActivityRef = new WeakReference<>(activity);
+            this.mReferrerGroup = referrerGroup;
+            this.isMarkingAsSpamInProgress = isMarkingAsSpamInProgress;
+        }
+
+        @Override
+        public void onResponse(final JSONObject response) {
+            if (mActivityRef.get() == null || mActivityRef.get().isFinishing()) {
+                return;
+            }
+
+            mReferrerGroup.isRestCallInProgress = false;
+            if (response!= null) {
+                boolean success = response.optBoolean("success");
+                if (success) {
+                    mReferrerGroup.isMarkedAsSpam = isMarkingAsSpamInProgress;
+                    int localBlogID = StatsUtils.getLocalBlogIdFromRemoteBlogId(
+                            Integer.parseInt(mReferrerGroup.getBlogId())
+                    );
+                    StatsTable.deleteStatsForBlog(mActivityRef.get(), localBlogID, StatsService.StatsEndpointsEnum.REFERRERS);
+                } else {
+                    // It's not a success. Something went wrong on the server
+                    String errorMessage = null;
+                    if (response.has("error")) {
+                        errorMessage = response.optString("message");
+                    }
+
+                    if (TextUtils.isEmpty(errorMessage)) {
+                        errorMessage = mActivityRef.get().getString(R.string.stats_referrers_spam_generic_error);
+                    }
+
+                    ToastUtils.showToast(mActivityRef.get(), errorMessage, ToastUtils.Duration.LONG);
+                }
+            }
+        }
+
+        @Override
+        public void onErrorResponse(final VolleyError volleyError) {
+            if (volleyError != null) {
+                AppLog.e(AppLog.T.STATS, "Error while marking the referrer " + getDomain(mReferrerGroup) + " as "
+                        + (isMarkingAsSpamInProgress ? " spam " : " unspam ") +
+                        volleyError.getMessage(), volleyError);
+            }
+            if (mActivityRef.get() == null || mActivityRef.get().isFinishing()) {
+                return;
+            }
+
+            mReferrerGroup.isRestCallInProgress = false;
+            ToastUtils.showToast(mActivityRef.get(),
+                    mActivityRef.get().getString(R.string.stats_referrers_spam_generic_error),
+                    ToastUtils.Duration.LONG);
```

Comment:
Given the length of the message, I'd say long, to be sure people have time to notice the message and read it before it disappears.